core/src/main/java/io/neow3j/transaction/Transaction.java

```diff
@@ -298,7 +294,6 @@ private void serializeWithoutWitnesses(BinaryWriter writer) throws IOException {
     @Override
     public void serialize(BinaryWriter writer) throws IOException {
         serializeWithoutWitnesses(writer);
-        this.witnesses.sort(Comparator.comparing(Witness::getScriptHash));
```

Comment:
Sorting obviously doesn't work if there are empty witnesses. I also didn't find the sorting anymore in the current neo code.

Reply:
very interesting...... so, if order doesn't matter, my comment above (about merging two empty witnesses) would make more or less sense? 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔

Reply:
Good point. The empty witnesses are anyway undistinguishable and could therefore be merged into one. I was not enlightened about this by the Neo code yet. I would need to read a bit more.